### PR TITLE
ANN: OpenResty 1.21.4.2 release.

### PR DESCRIPTION
**Pull request by xiaocang**

### CODE REVIEW

File: v2/cn/ann-1021004002.md.tt2 [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+# Version Highlights
+
+* [LuaJIT](https://github.com/openresty/luajit2#readme)
+    * Avoid negation of signed integers in C that may hold INT*_MIN, since recent C compilers 'take advantage' of the undefined behavior, this completely changes the meaning of expressions like `(k == -k)`.
+    * Correct fix for stack check when recording `BC_VARG`.
+    * Fix trace join to `BC_JLOOP` originating from `BC_ITERN`.
+    * Fix `math.floor()` and `math.ceil()`.
+    * Many fixes imported from Mike Pall's upstream https://github.com/luajit/luajit repository.
+* Fixed a bug that uses memory after free when encountering an invalid `IF-Match` request header.
```
Comment (dndx) [inline v2/cn/ann-1021004002.md.tt2:26]:
```suggestion
* Fixed a bug that uses memory after free when encountering an invalid `IF-Match` request header. Thanks to Michal Jeczalik Jr, Nishant Hooda and Kong Inc for reporting this issue.
```

To give credit for the reporters.

File: v2/en/ann-1021004002.md.tt2 [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+# Version Highlights
+
+* [LuaJIT](https://github.com/openresty/luajit2#readme)
+    * Avoid negation of signed integers in C that may hold INT*_MIN, since recent C compilers 'take advantage' of the undefined behavior, this completely changes the meaning of expressions like `(k == -k)`.
+    * Correct fix for stack check when recording `BC_VARG`.
+    * Fix trace join to `BC_JLOOP` originating from `BC_ITERN`.
+    * Fix `math.floor()` and `math.ceil()`.
+    * Many fixes imported from Mike Pall's upstream https://github.com/luajit/luajit repository.
+* Fixed a bug that uses memory after free when encountering an invalid `IF-Match` request header.
```
Comment (dndx) [inline v2/en/ann-1021004002.md.tt2:26]:
```suggestion
* Fixed a bug that uses memory after free when encountering an invalid `IF-Match` request header. Thanks to Michal Jeczalik Jr, Nishant Hooda and Kong Inc for reporting this issue.
```

File: v2/cn/ann-1021004002.md.tt2 [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+--->
+
+We are happy to announce the new formal release, [% version %], of the
+OpenResty web platform based on NGINX and LuaJIT.
+
+[Download this version here](download.html).
+
+The (portable) source code distribution, the Win32/Win64 binary
+distributions, and the pre-built binary Linux packages for Ubuntu,
+Debian, Fedora, CentOS, RHEL, Alpine, OpenSUSE, Amazon Linux, Rocky Linux,
```
Comment (zhuizhuhaomeng) [inline v2/cn/ann-1021004002.md.tt2:15]: Mariner

File: v2/en/ann-1021004002.md.tt2 [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+--->
+
+We are happy to announce the new formal release, [% version %], of the
+OpenResty web platform based on NGINX and LuaJIT.
+
+[Download this version here](download.html).
+
+The (portable) source code distribution, the Win32/Win64 binary
+distributions, and the pre-built binary Linux packages for Ubuntu,
+Debian, Fedora, CentOS, RHEL, Alpine, OpenSUSE, Amazon Linux, Rocky Linux,
```
Comment (zhuizhuhaomeng) [inline v2/en/ann-1021004002.md.tt2:15]: Ditto

**Review by zhuizhuhaomeng — APPROVED** [review]

**Pull request merged**